pkg/specs/apptype/determine_type.go

```diff
@@ -80,16 +81,35 @@ func (r *inspector) DetermineApplicationType(ctx context.Context, upstream strin
 		return &localAppCopy{AppType: "runbook.replicated.app", LocalPath: parts[0]}, nil
 	}
 
-	r.ui.Info(fmt.Sprintf("Attempting to retrieve upstream %s ...", upstream))
+	if i.viper.GetBool("isEdit") {
+		state, err := i.state.TryLoad()
+		if err != nil {
+			return nil, errors.Wrap(err, "load app state")
+		}
+		upstreamContents := state.UpstreamContents()
+		if upstreamContents == nil {
+			return nil, fmt.Errorf("no upstream contents present")
+		}
+
+		if upstreamContents.AppRelease != nil {
+			return &localAppCopy{AppType: "replicated.app"}, nil
+		}
+
+		// create a new fetcher class that gets things from the state file
+		stateClient := stategetter.NewStateGetter(i.fs, i.logger, upstreamContents)
```

Comment:
Okay, so it looks like StateGetter implements the same interface as the GoGetter client, to fetch things like Git artifacts and friends?

Reply:
The same interface as the GoGetter client, LocalGetter, and GithubClient

Reply:
Awesome 👍